Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/Module/DynamicModuleHandle_UnixLike.cpp

```diff
@@ -132,12 +131,18 @@ namespace AZ
             Unload();
         }
 
-        LoadStatus LoadModule(bool globalSymbols) override
+        LoadStatus LoadModule(LoadFlags flags) override
         {
             AZ::Debug::Trace::Instance().Printf("Module", "Attempting to load module:%s\n", m_fileName.c_str());
-            bool alreadyOpen = false;
 
-            m_handle = Platform::OpenModule(m_fileName, alreadyOpen, globalSymbols);
+            m_handle = dlopen(m_fileName.c_str(), RTLD_NOLOAD);
+            bool alreadyOpen = (m_handle != nullptr);
+            bool loadModule = !CheckBitsAny(flags, LoadFlags::NoLoad);
+            if (!alreadyOpen && loadModule)
+            {
+                int openFlags = CheckBitsAny(flags, LoadFlags::GlobalSymbols) ? RTLD_NOW | RTLD_GLOBAL : RTLD_NOW;
+                m_handle = dlopen(m_fileName.c_str(), openFlags);
+            }
```

Comment:
The `m_handle` variable will be set if `alreadyOpen` is `true`, so this would equal loading, even if `LoadFlags::NoLoad` is set, so it will report as loaded (`IsLoaded` checks `m_handle`).

Reply:
If the module is already loaded, then the first `dlopen` with the `RTLD_NOLOAD` parameter would retrieve that handle.
The logic looks correct, though a couple of lines could be saved by removing the boolean variables.

```suggestion
            if (m_handle = dlopen(m_fileName.c_str(), RTLD_NOLOAD); m_handle == nullptr && !CheckBitsAny(flags, LoadFlags::NoLoad))
            {
                int openFlags = CheckBitsAny(flags, LoadFlags::GlobalSymbols) ? RTLD_NOW | RTLD_GLOBAL : RTLD_NOW;
                m_handle = dlopen(m_fileName.c_str(), openFlags);
            }
```